### DISCUSSION TITLE
CLI command for removing all environment variables for specific branch?

### DISCUSSION THREAD

**Discussion by cceloso** [vercel/community#7423] [Help] [answered]
### Summary

Hello! Is there a CLI command available for removing all environment variables for a specific Git branch?

I can only find a command for removing one variable in one environment which is `vercel env rm [name] [environment]`.

I'm looking for something like this: `vercel env rm all [environment] [gitbranch].`

Thank you!

### Steps to Reproduce

1. Add environment variables for a certain Git branch in one environment, like preview.
2. Eventually the branch gets deleted, so I also need to delete the environment variables for it. Currently, I am doing this manually in the Vercel dashboard, but I'm hoping to find a better solution.

**Comment by pawlean** [comment]
Hi @cceloso!

Currently, Vercel does not offer a direct CLI command to remove all environment variables for a specific Git branch. See [relevant documentation.](https://vercel.com/docs/cli/env) I will pass this feature request internally. 

To manage this task, you could write a script that leverages the Vercel CLI to list and then individually remove environment variables associated with a particular branch.

Let us know how you get on!

**Comment by cceloso** [comment, marked as answer]
I see, thank you for creating a feature request @pawlean!

Since there is no way to target a specific Git branch yet, as a workaround, I have to remove all environment variables in an environment (in my case, preview environment), and then re-add them. (This might seem counterintuitive but I guess it's better than manually removing many variables? :smiling_face_with_tear:)

I've used the scripts presented [here](https://dev.to/hanksudo/batch-to-add-remove-environment-variables-by-vercel-cli-39nn) as a guide to create a custom script for my use case.

In my case, I have two branches associated with the preview environment, one for staging and another for a release candidate. I created a shell script together with an expect file, with the latter being used to automatically respond to CLI prompts.

Here's the shell script:

```
# Backup existing preview environment variables in staging (develop branch) and pre (old release candidate [rc] branch)
vercel env pull --environment=preview --git-branch=develop .env.staging
vercel env pull --environment=preview --git-branch=rc/v253 .env.pre

# Remove environment variables in preview environment
vercel env ls | sed 's/=/ /' | awk 'NR>2 && $3 == "Preview" {print $1, $3}' | xargs -o -n 2 './remove_vercel_env.exp'

# Re-add environment variables in preview environment from backup to staging (develop branch) and pre (new rc branch)
cat .env.staging | sed '1d' | sed 's/=/ /' | xargs -n 2 bash -c 'echo -n $1 | vercel env add $0 preview develop'
cat .env.pre | sed '1d' | sed 's/=/ /' | xargs -n 2 bash -c 'echo -n $1 | vercel env add $0 preview rc/v254'

# Optional: Remove backup
# rm .env.staging
# rm .env.pre
```

And here's the expect file for automating the response to CLI prompts:

```
#!/usr/bin/env expect

# Set script to run indefinitely until it's explicitly closed,
# and spawn the Vercel environment removal command
set timeout -1

# Arguments passed to the script
set var_name [lindex $argv 0]
set environment [lindex $argv 1]

# Convert environment to lowercase
set environment [string tolower $environment]

# Format the Vercel command to remove an environment variable,
# and bypass remove confirmation with --yes flag
spawn vercel env rm $var_name $environment --yes

# Turn on internal debugging
# exp_internal 1

# Optionally handle the choice of environment
expect {
    -re "Remove $var_name from which Environments\?" {
        # Press enter to select the default option
        send "\r"
        exp_continue
    }
    timeout {
        send_user "No environment selection prompt appeared.\n"
    }
}
```

[a comment by an automated account is not shown]
